packages/yarnpkg-pnpify/sources/sdks/base.ts

```diff
@@ -72,15 +72,29 @@ export const generateTypescriptBaseWrapper: GenerateBaseWrapper = async (pnpApi:
           str = str.replace(/\\\\/g, \`/\`)
           str = str.replace(/^\\/?/, \`/\`);
 
-          // Absolute VSCode \`Uri.fsPath\`s need to start with a slash.
-          // VSCode only adds it automatically for supported schemes,
-          // so we have to do it manually for the \`zip\` scheme.
-          // The path needs to start with a caret otherwise VSCode doesn't handle the protocol
-          //
-          // Ref: https://github.com/microsoft/vscode/issues/105014#issuecomment-686760910
-          //
           if (str.match(/\\.zip\\//)) {
-            str = \`\${isVSCode ? \`^\` : \`\`}zip:\${str}\`;
+            swtich (hostInfo) {
+              // Absolute VSCode \`Uri.fsPath\`s need to start with a slash.
+              // VSCode only adds it automatically for supported schemes,
+              // so we have to do it manually for the \`zip\` scheme.
+              // The path needs to start with a caret otherwise VSCode doesn't handle the protocol
+              //
+              // Ref: https://github.com/microsoft/vscode/issues/105014#issuecomment-686760910
+              //
+              case \`vscode\`:
+                str = \`^zip:\${str}\`; break;
+
+              // To make "go to definition work",
+              // resolve the actual file system path from virtual path
+              // and convert to scheme supported by [vim-rzip](https://github.com/lbrayner/vim-rzip)
+              case \`coc-nvim\`:
+                str = str.replace(/\\$\\$virtual\\/([\\-\\w\\/]+)\\/cache/, \`cache\`);
```

Comment:
The `resolved` variable contains the real path https://github.com/yarnpkg/berry/blob/b27cd5ebf40b2dbe80e81b7b3ba77b0edfe28ecb/packages/yarnpkg-pnpify/sources/sdks/base.ts#L64

Reply:
it works, thanks!